### PR TITLE
S3C-405: Trust policy validation

### PR DESCRIPTION
**Pull request by alexandre-merle**
Allowing trust policy to be evaluated

Fix https://scality.atlassian.net/browse/S3C-405

### CODE REVIEW

**Comment by alexandre-merle** [comment]
@ironman-machine try

File: lib/policy/userPolicySchema.json [hunk 1]
```diff
@@ -398,15 +519,31 @@
...
                         }
                     ]
                 }
             ]
         }
     },
-    "required": [
-        "Version",
-        "Statement"
+    "oneOf": [
```
Comment (rahulreddy) [inline lib/policy/userPolicySchema.json]: Why this change? Doesn't the earlier version work?

Comment (alexandre-merle) [inline lib/policy/userPolicySchema.json]: was some test i done previously, removed :)

File: tests/unit/policy/test_policyValidator.js [hunk 2]
```diff
@@ -71,6 +72,98 @@ describe('Policies validation - Version', () => {
...
+            value: {
+                Federated:
+                    'arn:aws:iam::111111111111:saml-provider/mysamlprovider',
+            },
+        },
+        {
+            name: 'with backbeat service',
+            value: { Service: 'backbeat' },
+        },
+    ].forEach(test => {
```
Comment (jonathan-gramain) [inline tests/unit/policy/test_policyValidator.js]: These tests are useful, though they should be completed with some failure tests to be really valuable.

Comment (alexandre-merle) [inline tests/unit/policy/test_policyValidator.js]: done :)

**Comment by alexandre-merle** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try
